### ISSUE TITLE
Can't access a bucket using the ${aws:username} as the bucket name

### ISSUE THREAD

**Issue by phasath** [minio/minio#14697]
tl;dr;:
I wanted to create buckets that has the same username name and provide a single policy that would work for all of them based on the username variable.

Full explanation:
I'm creating users and buckets with the same name. I have dozens of users/buckets. The users are created in the Identity -> Users. 

So let's say I create a user called `example` (accessKey = `example`). Then, I create one bucket called `example`, exactly as the user name created before. 

I want to create just one policy that should be generic enough, so I can assign the same policy to all users, using the username. This is the attempt I made, following the S3 guide and some other issues I found here. 
```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "s3:ListBucket"
            ],
            "Resource": [
                "arn:aws:s3:::${aws:username}",
                "arn:aws:s3:::${aws:username}/*"
            ]
        },
        {
            "Effect": "Allow",
            "Action": [
                "s3:GetObject",
                "s3:PutObject"
            ],
            "Resource": [
                "arn:aws:s3:::${aws:username}/*",
                 "arn:aws:s3:::${aws:username}"
            ]
        }
    ]
}
```
I also tried changing `${aws:username}` to `${jwt:prefered_username}`, `${ldap:userid}` and a bunch of other ways I found through search, but none of them seem to work. 

It is worth to point out that the `s3:ListBucket` works with the `arn:aws:s3:::${aws:username}` - I can login with the user and see the correct bucket (same name of the user). But when I click into `Browse`, it does nothing, does not allow me to access the bucket content.

If I simple add the bucket name on the resource for GetObject, it works, like: `"arn:aws:s3:::example/*"`, so the problem is only using the `${aws:usuername}`.

## Expected Behavior
I should be able to browse the content of the bucket with the given user.

## Current Behavior
It does not allow me to browse the content of the bucket.

## Steps to Reproduce (for bugs)
1. Fresh install of Minio (minio/minio:RELEASE.2022-04-01T03-41-39Z)
2. Create one user on identity
3. Create one bucket with the same name of the user 
4. Try to use the given policy so that the user can access that bucket.

## Context
I want to have this working so I can create multiple buckets / users and just have one policy for them, instead of having to create one policy for each user.

## Your Environment
* Docker release (minio/minio:RELEASE.2022-04-01T03-41-39Z)
* VM with 16gb RAM and 6 cores.
* Linux uploader-minio 5.4.0-107-generic #121-Ubuntu SMP Thu Mar 24 16:04:27 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux 
* Ubuntu 22.04 LTS Server
